server/src/main/java/org/opensearch/common/bytes/AbstractBytesReference.java

```diff
@@ -82,7 +84,13 @@ public void writeTo(OutputStream os) throws IOException {
 
     @Override
     public String utf8ToString() {
-        return toBytesRef().utf8ToString();
+        BytesRef bytesRef = toBytesRef();
+        final char[] ref = new char[bytesRef.length];
+        final int len = UnicodeUtil.UTF8toUTF16(bytesRef.bytes, bytesRef.offset, bytesRef.length, ref);
```

Comment:
nit : The wrapper method `UTF8toUTF16(BytesRef bytesRef, char[] chars)`, which handles passing the correct params, could also be used

Reply:
Good suggestion. Updated